### ISSUE TITLE
bug: iOS ITMS-90338: Non-public API usage

### ISSUE THREAD

**Issue by hussainarthuna** [ionic-team/capacitor#5661]
### Capacitor Version

```
💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 3.5.1
  @capacitor/core: 3.5.1
  @capacitor/android: 3.5.1
  @capacitor/ios: 3.5.1

Installed Dependencies:

  @capacitor/cli: 3.5.1
  @capacitor/core: 3.5.1
  @capacitor/ios: 3.5.1
  @capacitor/android: 3.5.1

[success] iOS looking great! 👌
[success] Android looking great! 👌
```

### Platform(s)

iOS

### Current Behavior

When Submitting the App to iOS the binary gets rejected and I receive the below email from Apple.

Dear Developer,

We identified one or more issues with a recent delivery for your app, "APP NAME" 1.0.3 (22). Please correct the following issues, then upload again.

ITMS-90338: Non-public API usage - The app references non-public selectors in Frameworks/Capacitor.framework/Capacitor: applicationNameForUserAgent, initWithFrame:configuration:, isMainFrame, setNavigationDelegate:, targetFrame. If method names in your source code match the private Apple APIs listed above, altering your method names will help prevent this app from being flagged in future submissions. In addition, note that one or more of the above APIs may be located in a static library that was included with your app. If so, they must be removed. For further information, visit the Technical Support Information at http://developer.apple.com/support/technical/

Best regards,

The App Store Team

### Expected Behavior

App Should be uploaded successfully

### Other Technical Details

`npm --version` output: 8.10.0

`node --version` output: v16.15.0

`pod --version` output (iOS issues only): 1.11.3

**Comment by paulovargatt** [comment]
same problem

**Comment by mlynch** [comment]
We're aware of this and unsure if this is a legitimate issue or a bug on Apple's end. Seeing reports of this in other ecosystems outside of Capacitor/Cordova

**Comment by mlynch** [comment]
Looks like it could possibly be an issue on Apple's side as these selectors are all public https://github.com/facebook/react-native/issues/33789#issuecomment-1147928476

**Comment by paulovargatt** [comment]
I believe apple is under maintenance https://developer.apple.com/system-status/

**Comment by hussainarthuna** [comment]
Thanks, @mlynch for the Quick response. Really Appreciate it.
Yes, I saw other platforms are reporting the same issue.

**Comment by pedrovmc** [comment]
Same on Flutter

**Comment by tonga54** [comment]
same problem

**Comment by masonicboom** [comment]
> We're aware of this and unsure if this is a legitimate issue or a bug on Apple's end. Seeing reports of this in other ecosystems outside of Capacitor/Cordova

@mlynch is Ionic working with Apple on this, or should individual devs escalate with Apple to get them focused on it?

**Comment by mlynch** [comment]
@masonicboom please escalate if you're able to, hearing from many developers is best!

**Comment by bazuka5801** [comment]
same problem

**Comment by dimarb** [comment]
same problem

**Comment by fohlin** [comment]
We're experiencing the same issue. Thanks the quick confirmation @mlynch and for everyone's attention on this.

**Comment by DanielPipa2** [comment]
same problem

**Comment by RRGT19** [comment]
Same

**Comment by lentyaishe** [comment]
Same problem here. 

Searched the resulting code by Ionic build for `applicationNameForUserAgent` and `isMainFrame` (other strings did not match anything) and all was found in WebViewDelegationHandler. Removed all the lines mentioning those (3 instances for `applicationNameForUserAgent` and 1 instance for `isMainFrame`) and the build magically got published.

Found on StackOverflow some thread from 2019 that also blamed (then) cordova-web-view-plugin..

Not sure about how bad is removing those lines of code.. But yet at least some sort of a solution if at all..

**Comment by rajauzair273** [comment]
same problem

**Comment by Jack-G-Smith** [comment]
+1 having this issue as well. does look to be linked to WebView usage, but that API is almost certainly public so I think it's a problem on Apple's side. Hopefully they fix it soon...

**Comment by mleusink** [comment]
Same problem

**Comment by stuartmorgan** [comment]
> that API is almost certainly public

There is no "almost" here; you can go read the public docs for these methods in Apple's public documentation, or in Apple's public headers that are part of the SDK. Where they have been since **iOS 8**.

**Comment by Jack-G-Smith** [comment]
Just saw this in their email about new requirements...

`Section 3.3.43: Specified requirements for use of the Apple Pay APIs with WKWebView.`

Might mean this is intended?

**Comment by beonde** [comment]
> Just saw this in Apple's recent email about new requirements...
> 
> `Section 3.3.43: Specified requirements for use of the Apple Pay APIs with WKWebView.`
> 
> Might mean this is intended?

Don't think so. We do not use any Apple Pay APIs. Also, the error being thrown by Apple is naming API calls that are not related to Apple Pay APIs.

**Comment by Jack-G-Smith** [comment]
> > Just saw this in Apple's recent email about new requirements...
> > `Section 3.3.43: Specified requirements for use of the Apple Pay APIs with WKWebView.`
> > Might mean this is intended?
> 
> Don't think so. We do not use any Apple Pay APIs. Also, the error being thrown by Apple is naming API calls that are not related to Apple Pay APIs.

yeah, deleted the previous comment for that reason, hoped to avoid misleading discussion. Hopefully they can get it together soon, going to open a support ticket now...

**Comment by trinhdh** [comment]
Same problem. So many of my clients are waiting on this release.

**Comment by georgekung** [comment]
Same issue using Cordova build. The iOS distribution/validator complain about the followings being non-public selectors: applicationNameForUserAgent, initWithFrame:configuration, isMainFrame, navigationDelegate, navigationType,
setNavigationDelegate, setProcessPool, targetFrame, userContentController.

Exact same code distributed fine yesterday.

**Comment by OtherCarlos** [comment]
Hi developers, I experience the same issue abode. Friday I upload an App successfully to TestFlight even today in the morning I upload the same App with a few changes on Typescript/Angular on my Capacitor App. Suddenly uploading a third minor change just 5 hours ago I get the same email over and over again: 

> ITMS-90338: Non-public API usage - The app references non-public selectors in Frameworks/Capacitor.framework/Capacitor: __swift_objectForKeyedSubscript:, applicationNameForUserAgent, initWithFrame:configuration:, isMainFrame, setNavigationDelegate:, targetFrame.

The funny thing it is that my Xcode show me Success uploaded App but App Connect reject me the version and send me the email.

Xcode 13.0
MacOs BigSur

**Comment by thuliumsystems** [comment]
same problem :(

**Comment by paulocoutinhox** [comment]
Hi,

Message from Apple: "The issue has been resolved on the app validation backend. Please try resubmitting. Sorry for the trouble."

Thanks.

**Comment by trinhdh** [comment]
I confirmed that It backs to work. I have just resubmitted 2 apps and it works as it should be.

Thanks for quick response.

**Comment by rdlabo** [comment]
me too!! Thanks!!

![スクリーンショット 2022-06-07 午後3 22 23](https://user-images.githubusercontent.com/9690024/172309973-109fb464-feea-4309-a21d-6488ac27e71f.png)

**Comment by jcesarmobile** [comment]
This was a problem on Apple backend, it’s fixed now

> "The issue has been resolved on the app validation backend. Please try resubmitting. Sorry for the trouble."


https://developer.apple.com/forums/thread/127678?answerId=715975022#715975022

**Comment by TannerPlauche** [comment]
Thank you for sharing and updating. 🙏 👍 ❤️

**Comment by jcesarmobile** [comment]
This was a problem on Apple servers and was fixed long ago (by Apple)
If you are facing a similar problem, report a new issue and provide a sample app so we can verify you are not using non public APIs in your app.
